openshift/app.dc.yaml

```diff
@@ -0,0 +1,166 @@
+---
+apiVersion: v1
+kind: Template
+labels:
+  app.kubernetes.io/component: app
+  app.kubernetes.io/instance: "${APP_NAME}-${JOB_NAME}"
+  app.kubernetes.io/name: nodejs
+  app.kubernetes.io/part-of: "${APP_NAME}-${JOB_NAME}"
+  app: "${APP_NAME}-${JOB_NAME}"
+  template: "${REPO_NAME}-app-dc-template"
+metadata:
+  name: "${REPO_NAME}-app-dc"
+objects:
+  - apiVersion: v1
+    kind: DeploymentConfig
+    metadata:
+      name: "${APP_NAME}-app-${JOB_NAME}"
+    spec:
+      replicas: 2
+      revisionHistoryLimit: 10
+      selector:
+        app: "${APP_NAME}-${JOB_NAME}"
+        deploymentconfig: "${APP_NAME}-app-${JOB_NAME}"
+        role: app
+      strategy:
+        type: Rolling
+        resources: {}
+        rollingParams:
+          timeoutSeconds: 600
+      template:
+        metadata:
+          labels:
+            app: "${APP_NAME}-${JOB_NAME}"
+            deploymentconfig: "${APP_NAME}-app-${JOB_NAME}"
+            role: app
+        spec:
+          containers:
+            - name: app
+              image: "${IMAGE_REGISTRY}/${APP_NAME}:${IMAGE_TAG}"
```

Comment:
This should be re-templated to `image: "${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/${IMAGE_NAME}:${IMAGE_TAG}"` or similar - lumping image registry with both the registry and the namespace as a default is bad practice.